routers/web/repo/http.go

```diff
@@ -609,3 +609,30 @@ func GetIdxFile(ctx *context.Context) {
 		h.sendFile("application/x-git-packed-objects-toc", "objects/pack/pack-"+ctx.Params("file")+".idx")
 	}
 }
+
+// GetAnnexObject implements git-annex dumb HTTP
+func GetAnnexObject(ctx *context.Context) {
```

Comment:
Okay so, almost half a year later, I finally looked into this. I am going to not do this.

LFS support seems to be kinda isolated. git-lfs specifies its own auth system using JWT so it can kinda live on its own. But here I call `httpBase()` which does auth enforcement -- and which is a module-private function. So it's just going to stay like it is.

Reply:
I think the JWT thing is why the LFS routes are pegged to

https://github.com/neuropoly/gitea/blob/2ec990931f61a1ef670b88c3bf37b06d4550548d/routers/web/web.go#L1206

while the rest of the HTTP git routes are pegged to

https://github.com/neuropoly/gitea/blob/2ec990931f61a1ef670b88c3bf37b06d4550548d/routers/web/web.go#L1222

It's because the LFS calls don't use the HTTP username so there's no need to assign it into the `ctx`.